File: src/CarDiaryX.Api/CarDiaryX.Application/Features/V1/Trips/InputModels/TripInputModel.cs
using System;

namespace CarDiaryX.Application.Features.V1.Trips.InputModels
{
    public class TripInputModel
    {
        public DateTimeOffset DepartureDate { get; set; }

        public DateTimeOffset ArrivalDate { get; set; }

        public AddressInputModel DepartureAddress { get; set; }

        public AddressInputModel ArrivalAddress { get; set; }

        public int? Distance { get; set; }

        public decimal? Cost { get; set; }

        public string Note { get; set; }

        public string RegistrationNumber { get; set; }
    }
}

File: src/CarDiary.Client/car-diary-x/src/app/core/models/trips/trip-input.model.ts
import { AddressModel } from './address.model';

export interface TripInputModel {
  departureDate: string;
  arrivalDate: string;
  departureAddress: AddressModel;
  arrivalAddress: AddressModel;
  registrationNumber: string,
  distance?: number;
  cost?: number;
  note?: string;
}
